packages/teams-js/src/public/media.ts

```diff
@@ -716,13 +719,7 @@ export namespace media {
       callback(oldPlatformError, null);
       return;
     }
-
-    if (!validateScanBarCodeInput(config)) {
-      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
-      callback(invalidInput, null);
-      return;
-    }
-
-    sendMessageToParent('media.scanBarCode', [config], callback);
+    config = config === undefined ? {} : config;
+    callCallbackWithErrorOrResultFromPromiseAndReturnPromise<string>(barCode.scanBarCode, callback, config);
```

Comment:
This function call should be reverted to just its old behavior and should no longer call the new barCode function. Barcode.isSupported will need to check both barcode and permissions, and it will be impossible to make this pattern work because permissions checking/requesting is totally new functionality.